pyupgrade/_plugins/typing_pep563.py

```diff
@@ -0,0 +1,107 @@
+import ast
+import functools
+from collections import deque
+from typing import Iterable
+from typing import List
+from typing import MutableSequence
+from typing import NamedTuple
+from typing import Optional
+from typing import Sequence
+from typing import Tuple
+
+from tokenize_rt import Offset
+
+from pyupgrade._ast_helpers import ast_to_offset
+from pyupgrade._data import register
+from pyupgrade._data import State
+from pyupgrade._data import Token
+from pyupgrade._data import TokenFunc
+
+
+class Replacement(NamedTuple):
+    node: ast.expr
+    new: str
+
+
+def _supported_version(state: State) -> bool:
+    return (
+        state.settings.min_version >= (3, 10) or (
+            'annotations' in state.from_imports['__future__']
+        )
+    )
+
+
+def _dequote(i: int, tokens: List[Token], *, new: str) -> None:
+    tokens[i] = tokens[i]._replace(src=new)
+
+
+def _replace_string_literal(
+    annotation: Optional[ast.expr],
+) -> Sequence[Replacement]:
+    nodes: MutableSequence[ast.AST] = deque()
+    if annotation is not None:
+        nodes.append(annotation)
+    replacements: MutableSequence[Replacement] = deque()
+
+    while nodes:
+        node = nodes.pop()
+        if isinstance(node, ast.Subscript):
+            if isinstance(node.value, ast.Name):
+                if node.value.id not in {
+                    'Literal',
+                    'NamedTuple',
+                    'TypedDict',
+                    'Annotated',
+                }:
+                    nodes.append(node.slice)
```

Comment:
this breaks for `typing.Annotated["foo", "bar"]`

also there are parts of those which are valid types that should be unquoted -- just not the whole thing

also for `NamedTuple` and `TypedDict` those aren't `Subscripts` those'll be calls

here's probably a useful testcase

```python
class D(TypedDict):
    E: TypedDict('E', {"foo": "int"})
```